src/core/mount.c

```diff
@@ -1053,11 +1053,14 @@ static void mount_enter_mounting(Mount *m) {
         if (p && mount_is_bind(p)) {
                 r = mkdir_p_label(p->what, m->directory_mode);
                 /* mkdir_p_label() can return -EEXIST if the target path exists and is not a directory - which is
-                 * totally OK, in case the user wants us to overmount a non-directory inode. */
-                if (r < 0 && r != -EEXIST) {
-                        log_unit_error_errno(UNIT(m), r, "Failed to make bind mount source '%s': %m", p->what);
-                        goto fail;
-                }
+                 * totally OK, in case the user wants us to overmount a non-directory inode. Also -EROFS can be
+                 * returned on read-only filesystem. Moreover, -EACCES (and also maybe -EPERM?) may be returned
+                 * when the path is on NFS and the kernel cache about the path is cold. See issue #24120.
```

Comment:
i'd drop the comment about the kernel cache. I think what matters is that on Linux if we try to create an inode that already exists and also lack the perms to do it, then EEXIST generally trumps EACCES. However, on NFS servers might do it the other way. Hence we need to handle both permission issues and EEXIST issues the same way.

Reply:
OK. Dropped.